Feature/diferent file name

magrathj opened the pull request:
Commit to address Issues: #26  - added feature to saveDataFlatfile function for #31

magrathj committed:
type: feature

Adding feature to enable changing of the flat's filename. Addressing issue @31

"export with a diferent file name #31"

magrathj committed:
Update shinyform.R

added changes suggested by daattali in previous pull request

daattali commented on R/shinyform.R:
```diff
@@ -79,21 +79,26 @@ loadData <- function(storage) {
-    ),
-    ".csv"
-  )
+  if (!is.null(storage$filename)) {
```
It's more readable to start with the `is.null()` case rather than the `!is.null()`, it's just easier on the brain to do positives than negatives. And since the if statement is really just determining the base filename (not the extension), it's a bit simpler to do this:

```
if (is.null(...) {
  filename <- paste(...)
} else {
  filename <- storage$filename
}
filename <- paste0(filename, ".csv")
```

magrathj replied:
thanks for the feedback

magrathj committed:
type: fix

Thanks for the feedback, makes sense.  Made changes from the review

daattali commented:
Please **always** test every line of code that gets modified. Even very small and innocent-looking changes **must** be tested before committing. 

The variables `fileName` vs `filename` will cause this code to break

magrathj committed:
Merge branch 'master' into feature/diferent_file_name

magrathj committed:
type: test

Added test file for testsavedataflatfile function to run unit tests on the function (all passed) and re-ran devtool::test() and devtool::document()
